matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/network/Request.kt

```diff
@@ -74,23 +84,16 @@ internal suspend inline fun <DATA> executeRequest(globalErrorReceiver: GlobalErr
 
             currentRetryCount++
 
-            if (exception is Failure.ServerError &&
-                    exception.httpCode == 429 &&
-                    exception.error.code == MatrixError.M_LIMIT_EXCEEDED &&
-                    currentRetryCount < maxRetriesCount) {
-                // 429, we can retry
-                delay(exception.getRetryDelay(1_000))
-            } else if (canRetry && currentRetryCount < maxRetriesCount && exception.shouldBeRetried()) {
-                delay(currentDelay)
+            if (canRetryOnFailure(exception) && currentRetryCount < maxRetriesCount) {
+                delay(exception.getRetryDelay(currentDelay))
```

Comment:
This is not strictly equivalent with the previous version, when we had a 429, the default delay was 1s. But I think it is still fine.

Reply:
Yes indeed, if we want to keep previous behaviour for 429, we should also return a fixed delay in the lambda (or null to use the default delay behaviour which is multiplied by 2 after each failure).

With my current implementation, receiving 4 successive 429 errors will lead to a total delay of 15 seconds (1 + 2 + 4 +8) instead of 4 seconds with the fixed delay. We should probably have to think about it. Was the previous behaviour originally expected for this special case ?

Note that the delay is passed as a default value in this special case, the delay passed in the 429 error body should be used, if any (it is not a mandatory field).

WDYT @bmarty?

Reply:
I see, the point is that if you want to have an automatic retry in case of 429, except if the delay is too long (which could be configurable by the app), it will be more difficult with your change.
So maybe keep the two cases (429 and canRetry) clearly separated.

Reply:
Ack. I close this PR and will create a new one.